frontend/src/components/BudgetLineItems/BLIReviewTable/BLIReviewTable.jsx

```diff
@@ -47,12 +47,12 @@ const AgreementBLIReviewTable = ({
         <th>
             <input
                 className="usa-checkbox__input"
-                id={`check-all-${servicesComponentId}`} // Use unique ID
+                id={`check-all-${servicesComponentNumber}`}
```

Comment:
Not sure if this will work for historical data: If `servicesComponentNumber` is equivalent to `services_component.number` this is not unique (at least in the historical data). You can have e.g. 3 services components with:

| id | number | sub_component |
|----|--------|---------------|
| 1  | 1      | 1.A           |
| 2  | 1      | 1.B           |
| 3  | 1      | 1.C           |